supervisor/docker/__init__.py

```diff
@@ -265,6 +282,42 @@ def repair(self) -> None:
         except docker.errors.APIError as err:
             _LOGGER.warning("Error for networks prune: %s", err)
 
+        _LOGGER.info("Fix stale container on hassio network")
+        try:
+            self.prune_networks(DOCKER_NETWORK)
+        except docker.errors.APIError as err:
+            _LOGGER.warning("Error for networks hassio prune: %s", err)
+
+        _LOGGER.info("Fix stale container on host network")
+        try:
+            self.prune_networks(DOCKER_NETWORK_HOST)
+        except docker.errors.APIError as err:
+            _LOGGER.warning("Error for networks host prune: %s", err)
+
+    def prune_networks(self, network_name: str) -> None:
+        """Prune stale container from network.
+
+        Fix: https://github.com/moby/moby/issues/23302
+        """
+        network: docker.models.networks.Network = self.docker.networks.get(network_name)
```

Comment:
Add a check here to catch if `network` is `None`, log a warning and return if network is not found.

Reply:
They can't be None, It fails which is covered by above